### PR TITLE
Linked API to Create School button, added tests and Storybook

### LINKED ISSUE #7: Make the "create school" button work.

### LINKED ISSUE #8: Link the create school button to correct API and make a SchoolCreate.js page for that.

### PR DESCRIPTION
**Pull request by sherpatran**
Made sure that the CreateSchool button is working. Connected it to the proper Schools POST api and added the tests/storybook for that feature. 

The parameter termRegex had to be encoded because of the disallowed characters "[" and "\\". This allows termRegex to be passed through the frontend without any issues. Then, to get the accurate value passed into the backend, termRegex is decoded in the SchoolController.

https://organic-sherpatran-dev.dokku-11.cs.ucsb.edu/

Closes #7 and closes #8.

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/organic/Interceptors/RoleUserInterceptor.java [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package edu.ucsb.cs156.organic.interceptors;
+package edu.ucsb.cs156.organic.Interceptors;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/organic/Interceptors/RoleUserInterceptor.java]: NOOOOO!!! 

We already fixed this!
Don't bring back this bad code that is unrelated to your changes!

File: src/main/java/edu/ucsb/cs156/organic/Interceptors/RoleUserInterceptorAppConfig.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package edu.ucsb.cs156.organic.interceptors;
+package edu.ucsb.cs156.organic.Interceptors;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/organic/Interceptors/RoleUserInterceptorAppConfig.java]: NOOOOO!!! 

We already fixed this!
Don't bring back this bad code that is unrelated to your changes!

File: src/test/java/edu/ucsb/cs156/organic/Interceptors/RoleUserInterceptorTests.java [hunk 3]
```diff
@@ -1,5 +1,5 @@
 
-package edu.ucsb.cs156.organic.interceptors;
+package edu.ucsb.cs156.organic.Interceptors;
```
Comment (pconrad) [inline src/test/java/edu/ucsb/cs156/organic/Interceptors/RoleUserInterceptorTests.java]: NOOOOO!!! 

We already fixed this!
Don't bring back this bad code that is unrelated to your changes!

File: src/main/java/edu/ucsb/cs156/organic/controllers/SchoolController.java [hunk 4]
```diff
@@ -108,21 +107,22 @@ public Object deleteSchool(
...
     public School postSchool(
         @Parameter(name="abbrev", description="university domain name", example="ucsb") @RequestParam String abbrev,
         @Parameter(name="name", description="University name") @RequestParam String name,
         @Parameter(name="termRegex", description="Format: Example [WSMF]\\d\\d") @RequestParam String termRegex,
         @Parameter(name="termDescription", description="Enter quarter, e.g. F23, W24, S24, M24") @RequestParam String termDescription,
         @Parameter(name="termError", description="input error?") @RequestParam String termError)
-        {
+        throws JsonProcessingException {
+
+        String decodedTermRegex = URLDecoder.decode(termRegex, StandardCharsets.UTF_8);
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/organic/controllers/SchoolController.java:119]: This should be discussed in the PR description.

**Review by pconrad — CHANGES_REQUESTED** [review]
Several things that need to be fixed...

**Review by 30912hyl — CHANGES_REQUESTED** [review]
Can you add me as admin for testing dev deployment?

**Review by pconrad — DISMISSED** [review]
image

**Review by pconrad — CHANGES_REQUESTED** [review]
Why are there changes to backend files in this PR?

It makes this harder to review.  Including changes to files that are unrelated to the PR is frowned upon in professional SW dev.

This page explains how you can easily remove the changes to some files from your PR:

https://ucsb-cs156.github.io/topics/pull_requests/pull_requests_simplifying.html

**Comment by sherpatran** [comment]
> Why are there changes to backend files in this PR?
> 
> It makes this harder to review. Including changes to files that are unrelated to the PR is frowned upon in professional SW dev.
> 
> This page explains how you can easily remove the changes to some files from your PR:
> 
> https://ucsb-cs156.github.io/topics/pull_requests/pull_requests_simplifying.html

The changes to the backend were necessary because of the encoding issues with TermRegex. I also believe that I removed the changes to the interceptor. Should I still break up this PR into the changes to frontend and backend?

**Comment by sherpatran** [comment]
Created new PR with cleaner commits.

**Pull request closed without merging**